views-pebble/src/main/java/io/micronaut/views/pebble/PebbleViewsRenderer.java

```diff
@@ -61,7 +61,7 @@ public PebbleViewsRenderer(PebbleConfiguration configuration, PebbleEngine engin
 
     @Override
     public Writable render(String name, T data, @NonNull HttpRequest<?> request) {
-        return (writer) -> engine.getTemplate(name).evaluate(writer, ViewUtils.modelOf(data));
+        return (writer) -> engine.getTemplate(name).evaluate(writer, ViewUtils.modelOf(data), request.getLocale().orElse(null));
```

Comment:
Please, use the [LocaleResolver](https://docs.micronaut.io/latest/guide/#localeResolution) api instead.

Reply:
Thank you. I tried to use it the same way `ThymeleafViewsRenderer` does.

I'm not too happy about the way in the deprecated constructor though...what do you think?
Should the anonymous `HttpLocaleResolver` be extracted to it's own class to avoid duplication?

Reply:
@andreas-mausch please, if you found other places where we are not resolving locale properly submit a PR. You help is really welcomed.

Reply:
@sdelamo Sure. I appreciate your lightning fast feedback and merge. That was fun.

I have two things (not locale-related) self-written for Micronaut+Pebble:

- Custom pebble function `message` to map i18n calls to ResourceBundleMessageSource
- Another custom function `asset` which prefixes a file with the Micronaut context-path.

But I'm not sure if I can generalize it easily. When I come up with something, I will contact you.